File: TwoSum.py
class Solution(object):
	def __init__(self):
		super(Solution, self).__init__()

	# @return a tuple, (index1, index2)
	def twoSum(self, num, target):
		num_sorted = sorted(num)
		# print num_sorted
		# print target
		begin = 0
		end = len(num)-1
		while ((num_sorted[begin]+num_sorted[end]) != target) and (begin < end):
			# print begin,end,num_sorted[begin],num_sorted[end]
			if num_sorted[begin]+num_sorted[end] >target:
				end = end-1
			elif num_sorted[begin] + num_sorted[end] <target:
				begin = begin +1

		# print begin,end,num_sorted[begin],num_sorted[end]
		ret1 = num.index(num_sorted[begin])+1
		del num[ret1-1]
		ret2 = num.index(num_sorted[end])+1
		if ret1<ret2:
			ret2= ret2+1
		else:
			ret = ret1
			ret1 = ret2
			ret2 = ret
		return (ret1,ret2)

def main():
	while True:
		number, target = raw_input().split(' ')
		# print number, target
		
		
		numbers = raw_input().split(' ')
		for x in xrange(0, len(numbers)):
			numbers[x]= int(numbers[x])
		a = Solution()
		print a.twoSum(numbers,int(target))
	

if __name__ == '__main__':
	main()

File: TwoSum.cpp
#include <vector>
#include <string.h>
#include <iostream>
using namespace std;
class Solution {
	public:
		vector<int> twoSum(vector<int> &numbers, int target) {
			std::vector<int> copy;
			copy = numbers;
			this->qsort(numbers, 0, numbers.size()-1);
			int i = 0, j = numbers.size()-1;
			while(numbers[i]+numbers[j] != target && i<j)
			{
				if (numbers[i]+numbers[j]<target && i<j)
				{
					i++;
					/* code */
				}
				if (numbers[i]+numbers[j]>target && i<j)
				{
					j--;
					/* code */
				}
			}
			std::vector<int> v;
			for (int x = 0; x < copy.size(); ++x)
			{
				if(copy[x]== numbers[i] || copy[x]== numbers[j])
					v.push_back(x+1);
				/* code */
			}
        		return v;
		}

		void qsort(std::vector<int>& numbers, int begin, int end){

			int number = numbers[(begin+end)/2];
			int i = begin;
			int j = end; 
			while(i<=j)
			{
				while(numbers[i]<number && i<=j){
					i++;
				}
				while(numbers[j]>number && i<=j){
					j--;
				}
				if(i<=j)
				{
					int temp = numbers[i];
					numbers[i] = numbers[j];
					numbers[j] = temp;
					i++;
					j--;	
				}
			}
			if(begin<j)qsort(numbers, begin, j);
			if(i<end)qsort(numbers, i, end);
		}
};

int main(int argc, char const *argv[])
{
	Solution solution;
	std::vector<int> v;
	int t,target;
	cin>>t>>target;
	for (int i = 0; i < t; ++i)
	{
		int j;
		cin>>j;
		v.push_back(j);
		/* code */
	}
	std::vector<int> newv = solution.twoSum(v, target);
	for (int i = 0; i < newv.size(); ++i)
	{
		cout<<newv[i]<<' ';
		/* code */
	}
	cout<<endl;	
	
	return 0;
}